src/network/networkctl.c

```diff
@@ -1067,7 +1071,93 @@ static int dump_statistics(Table *table, const LinkInfo *info) {
         return 0;
 }
 
+static OutputFlags get_output_flags(void) {
+        return
+                arg_all * OUTPUT_SHOW_ALL |
+                (arg_full || !on_tty() || pager_have()) * OUTPUT_FULL_WIDTH |
+                colors_enabled() * OUTPUT_COLOR;
+}
+
+static int show_logs(sd_bus *bus, const char *ifname) {
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message *reply = NULL;
+        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_(sd_journal_closep) sd_journal *j = NULL;
+        _cleanup_free_ char *path = NULL;
+        usec_t inactive;
+        int r;
+
+        assert(bus);
+
+        if (arg_lines == 0)
+                return 0;
+
+        path = unit_dbus_path_from_name("systemd-networkd.service");
+        if (!path)
+                return log_oom();
+
+        r = sd_bus_call_method(
+                        bus,
+                        "org.freedesktop.systemd1",
+                        path,
+                        "org.freedesktop.DBus.Properties",
+                        "Get",
+                        &error,
+                        &reply,
+                        "ss",
+                        "org.freedesktop.systemd1.Unit",
+                        "InactiveExitTimestampMonotonic");
+        if (r < 0)
+                return log_error_errno(r, "Failed to get properties: %s", bus_error_message(&error, r));
+
+        r = sd_bus_message_enter_container(reply, 'v', "t");
+        if (r < 0)
+                return bus_log_parse_error(r);
+
+        r = sd_bus_message_read(reply, "t", &inactive);
+        if (r < 0)
+                return bus_log_parse_error(r);
+
+        r = sd_bus_message_exit_container(reply);
+        if (r < 0)
+                return bus_log_parse_error(r);
+
+        r = sd_journal_open(&j, SD_JOURNAL_LOCAL_ONLY);
+        if (r < 0)
+                return log_error_errno(r, "Failed to open journal: %m");
+
+        r = add_match_this_boot(j, NULL);
+        if (r < 0)
+                return log_error_errno(r, "Failed to add boot matches: %m");
+
+        if (ifname) {
+                const char *m;
+
+                m = strjoina("INTERFACE=", ifname);
+                (void)(
+                       (r = sd_journal_add_match(j, "_SYSTEMD_UNIT=systemd-networkd.service", 0)) ||
+                       (r = sd_journal_add_match(j, m, 0))
+                );
+                if (r < 0)
+                        return log_error_errno(r, "Failed to add link matches: %m");
+        } else {
+                r = add_matches_for_unit(j, "systemd-networkd.service");
+                if (r < 0)
+                        return log_error_errno(r, "Failed to add unit matches: %m");
+        }
```

Comment:
hmm, I think we should simplify this, and just log everything matching, without limiting things to networkd or so...

i.e. networkctl should show *everything* about the interfcae, regardless who logs that. For example, if openvpn or openvswitch or so manage an iface together with networkd we should show the output of all of them, interleaved with networkd's output (well, only if they are willing to augment their logs with journald metadata). i.e. let's be comphrensive here: if people specify INTERFACE=, then that's what counts, regardless who does it... 

(Admittedly INTERFACE is a bit generically named, a property of NETIF= or so IFNAME= or so might be better, maybe we can still get away with changing that?)

(and we probably should also show logs of `_KERNEL_DEVICE=n4711` when called for an interface with ifindex 4711, so that the kernel output tagged properly shows up too)